compiler/src/dmd/backend/mscoffobj.d

```diff
@@ -649,8 +650,23 @@ void MsCoffObj_term(const(char)[] objfilename)
     BIGOBJ_HEADER header = void;
     IMAGE_FILE_HEADER header_old = void;
 
-    time_t f_timedat = 0;
-    time(&f_timedat);
+    // Use a hash of segment content instead of current timestamp for reproducible builds
+    ubyte[] hashInput;
+    for (segidx_t seg = 1; seg < SegData.length; seg++)
+    {
+        seg_data* pseg = SegData[seg];
+        if (pseg.SDbuf && pseg.SDbuf.length())
+        {
+            // Sample a few bytes from each segment
+            size_t bytesToAdd = pseg.SDbuf.length() >= 16 ? 16 : pseg.SDbuf.length();
+            hashInput.length += bytesToAdd;
+            memcpy(&hashInput[hashInput.length - bytesToAdd], pseg.SDbuf.buf, bytesToAdd);
```

Comment:
Hashing doesn't require allocating a buffer. You can hash the result of each section and combine the hash results using something like dmd.root.hash.mixHash. (Although the dmd.root package can't be used by the backend, so perhaps you can move it to dmd.common?)